### ISSUE TITLE
Add option to save default value to `use_storage`

### ISSUE THREAD

**Issue by maccesch** [Synphonyte/leptos-use#123]

**Comment by mscofield0** [comment]
I'm confused, doesn't it already save the default value?
In `src/storage/use_storage.rs`:
```rust
        let fetch_from_storage = {
            let storage = storage.to_owned();
            let key = key.as_ref().to_owned();
            let on_error = on_error.to_owned();

            move || {
                let fetched = storage
                    .to_owned()
                    .and_then(|storage| {
                        // Get directly from storage
                        let result = storage
                            .get_item(&key)
                            .map_err(UseStorageError::GetItemFailed);
                        handle_error(&on_error, result)
                    })
                    .unwrap_or_default() // Drop handled Err(()) # <--- Default gets generated and saved regardless
                    .as_ref()
                    .map(|encoded| {
                        // Decode item
                        let result = C::decode(encoded)
                            .map_err(|e| UseStorageError::ItemCodecError(CodecError::Decode(e)));
                        handle_error(&on_error, result)
                    })
                    .transpose()
                    .unwrap_or_default(); // Drop handled Err(())

                match fetched {
                    Some(value) => {
                        // Replace data if changed
                        if value != data.get_untracked() {
                            set_data.set(value)
                        }
                    }

                    // Revert to default
                    None => set_data.set(default.clone()),
                };
            }
        };
```

If you mean conditionalizing that feature, what scenario are you thinking of? Seems like it's never useful to *not* save the default value.

**Comment by maccesch** [comment]
Hm you're half right. While indeed the `data` signal is set this doesn't always mean that it is also written to storage. The `watch` to do that is executed after the first call to `fetch_from_storage`. Although that is not the case when hydrating. 

So we have some inconsistent behavior here. I guess for now it makes sense to always write it and see if someone doesn't want that behavior

**Comment by maccesch** [comment]
Fixed in c26a96d8a671497ffb1613fb6b10d52dea8edc4e
